docs/docs/authentication/google-sso.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: default
+title: Google
+nav_order: 1
+parent: Authentication
+description: >
+  This page contains instructions on setting up Google
+  as the SSO provider for Firezone.
+---
+---
+
+Firezone supports Single Sign-On (SSO) through Google Workspace.
+After successfully configuring SSO with Firezone, users will be prompted to sign
+in with their Google credentials to in the Firezone portal to authenticate VPN
+sessions, and download device configuration files.
+
+![Firezone Google SSO Login](){:width="600"}
+
+To set up SSO, follow the steps below:
+
+## Step 1 - Create OAuth Client IDs
+
+_This section is based off Google's own documentation on
+[setting up OAuth 2.0](https://support.google.com/cloud/answer/6158849)._
+
+Visit the Google Cloud Console
+[Credentials page](https://console.cloud.google.com/apis/credentials)
+page, click `+ Create Credentials` and select `OAuth client ID`.
+
+![Create OAuth Client ID](https://user-images.githubusercontent.com/52545545/155904211-c36095b9-4bbd-44c1-95f8-bb165e314af3.png){:width="800"}
+
+On the OAuth client ID creation screen:
+
+1. Set `Application Type` to `Web application`
+1. Add an entry to Authorized redirect URIs that consists of appending
+`/auth/google/callback` to your Firezone base URL. For example, if your Firezone
+instance is available at `https://firezone.example.com`, then you would enter
+`https://firezone.example.com/auth/google/callback` here. The redirect URI is
+where Google will redirect the user's browser after successful authentication.
+Firezone will receive this callback, initiate the user's session, and redirect
+the user's browser to the appropriate page depending on the user's role.
+
+![Create OAuth client ID](https://user-images.githubusercontent.com/52545545/155904581-9a82fc9f-26ce-4fdf-8143-060cbad0a207.png){:width="800"}
+
+Note: if this is the first time you are creating a new OAuth client ID, you will
+be asked to configure a consent screen. Configuring the consent screen is outside
+the scope of this guide, see
+[Google's documentation](https://support.google.com/cloud/answer/10311615)
+for more information.
+
+After creating the OAuth client ID, you will be given a Client ID and Client Secret.
+These will be used in Step 2.
+![Copy Client ID and Secret](https://user-images.githubusercontent.com/52545545/155906344-aa3673e1-903a-482f-86fb-75f12fd17f4f.png){:width="800"}
+
+## Step 2 - Configure Firezone
```

Comment:
Instructions:

Using the client ID, secret, and redirect URI from above, edit the `/etc/firezone/firezone.rb` configuration file to include the following options:

```ruby
default['firezone']['authentication']['google']['enabeld'] = true
default['firezone']['authentication']['google']['client_id'] = 'CLIENT_ID'
default['firezone']['authentication']['google']['client_secret'] = 'CLIENT_SECRET'
default['firezone']['authentication']['google']['redirect_uri'] = 'REDIRECT_URI'
```

Run the following commands to apply the changes:

```bash
sudo firezone-ctl reconfigure
sudo firezone-ctl restart
```

You should now see a `Sign in with Google` button at the root Firezone URL.

Reply:
also did this for Okta